Makefile

```diff
@@ -51,19 +51,40 @@ export: VERSION.full
 	rm bats.tar
 	ls -lh charliecloud-$$(cat VERSION.full).tar.gz
 
-BIN=$(PREFIX)/bin
-DOC=$(PREFIX)/share/doc/charliecloud
+INSTALL_PREFIX := $(if $(DESTDIR),$(DESTDIR),$(PREFIX))
+BIN=$(INSTALL_PREFIX)/bin
+LIBEXEC=$(INSTALL_PREFIX)/libexec/charliecloud
+LIBEXEC_RUN=$(PREFIX)/libexec/charliecloud
+DOC=$(INSTALL_PREFIX)/share/doc/charliecloud
 TEST=$(DOC)/test
 .PHONY: install
 install: all
-	@test -n "$(PREFIX)" || \
+	@test -n "$(INSTALL_PREFIX)" || \
           (echo "No PREFIX specified. Lasciando ogni speranza." && false)
-	@echo Installing in $(PREFIX)
+	@echo Installing in $(INSTALL_PREFIX)
 #       binaries
 	install -d $(BIN)
 	install -pm 755 -t $(BIN) $$(find bin -type f -executable)
-	if [ -u bin/ch-run ]; then sudo chmod u+s $(BIN)/ch-run; fi
-	install -pm 644 -t $(BIN) bin/base.sh bin/version.h bin/version.sh
+	# Modify scripts to relate to new libexec location
+	for scriptfile in $$(find bin -type f -executable -printf "%f\n"); do \
+	    sed -i "s#^libexec_base=.*#libexec_base=$(LIBEXEC_RUN)#" $(BIN)/$${scriptfile}; \
+	done
```

Comment:
It would be nice if we didn't have to patch the scripts at install time. Is this a best practice? Are there alternatives?

Reply:
I'm not aware of good alternatives. Most packages avoid this issue by not being runnable from the build location, so they directly store the libexec-path inside the scripts - or, they use autotools / cmake, have something like `@LIBEXEC@` in the files and let autotools / cmake do the replacing of that placeholder. 
If you (or anybody else) finds something nicer, I'm all open ears - I'm also not really a fan of using `sed` here ;-).

Reply:
I also gave it a thought again and remembered another approach, I'll mention it just for completeness, since it does not offer significant advantages. 
What some programs do, is to have an extra binary (such as `root-config`, `geant4-config`) installed in the path which answers such questions for paths at runtime, i.e. `root-config --exec-prefix`, or also version information, such as `root-config --version`. The way that binary is generated usually involves the same technique, though: autotools or cmake replace macros in some header file, which is then included in the compiled binary. That's commonly used for finding libraries and `LDFLAGS` and things like that (alternatively, `pkgconfig` is used). 
So I don't see any obvious advantage to that - the only gain would be there is one place where paths are hardcoded at install-time, and not each script separately.